demo/clean-kubernetes.sh

```diff
@@ -5,14 +5,22 @@ set -aueo pipefail
 # shellcheck disable=SC1091
 source .env
 
+MESH_NAME="${MESH_NAME:-osm}"
+K8S_NAMESPACE="${K8S_NAMESPACE:-osm-system}"
+BOOKBUYER_NAMESPACE="${BOOKBUYER_NAMESPACE:-bookbuyer}"
+BOOKSTORE_NAMESPACE="${BOOKSTORE_NAMESPACE:-bookstore}"
+BOOKTHIEF_NAMESPACE="${BOOKTHIEF_NAMESPACE:-bookthief}"
+BOOKWAREHOUSE_NAMESPACE="${BOOKWAREHOUSE_NAMESPACE:-bookwarehouse}"
+TIMEOUT="${TIMEOUT:-90s}"
+
 bin/osm uninstall -f --mesh-name "$MESH_NAME" --osm-namespace "$K8S_NAMESPACE"
 
 for ns in "$BOOKWAREHOUSE_NAMESPACE" "$BOOKBUYER_NAMESPACE" "$BOOKSTORE_NAMESPACE" "$BOOKTHIEF_NAMESPACE" "$K8S_NAMESPACE"; do
-    kubectl delete namespace "$ns" --ignore-not-found --wait &
+    kubectl delete namespace "$ns" --ignore-not-found --wait --timeout="$TIMEOUT" &
 done
 
 # Clean up Hashicorp Vault deployment
-kubectl delete deployment vault -n "$K8S_NAMESPACE" --ignore-not-found --wait &
-kubectl delete service vault -n "$K8S_NAMESPACE" --ignore-not-found --wait &
+kubectl delete deployment vault -n "$K8S_NAMESPACE" --ignore-not-found --wait --timeout="$TIMEOUT" &
```

Comment:
What's the default timeout, and what happens if the configures timeout exceeds?

Reply:
from kubectl delete --help:

```
--timeout=0s: The length of time to wait before giving up on a delete, zero means determine a timeout from the
size of the object
```

However, I have found this is not reliable and when running the automated demo from CI on openshift, the delete command returns but the namespace is still in a "deleting state" despite the --wait flag being supplied. So the next step in the demo (to create the ns) fails.